bin/install_cocoapods

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -e
+
+# Start by restoring specs repo and pod cache:
+# - The specs repo cache holds all of the Podspec files. This avoids having download them all from the CDN.
+# - The pod cache holds the downloaded Pod source files. This avoids having to check them out again.
+restore_cache "$BUILDKITE_PIPELINE_SLUG-specs-repos"
+restore_cache "$BUILDKITE_PIPELINE_SLUG-global-pod-cache"
+
+# Restore the local `Pods` directory based on the `Podfile.lock` contents
+restore_cache "$(hash_file Podfile.lock)"
+
+# If the `pod check` plugin is installed, use it to determine whether or not to install Pods at all
+# If it's not installed (or if it fails), we'll try to install Pods. 
+# If that fails, it may be due to an out-of-date repo. We can use `--repo-update` to try to resolve this automatically.
+if [ $(bundle exec pod plugins installed | grep check) ]; then
+	bundle exec pod check || bundle exec pod install || bundle exec pod install --repo-update --verbose
+else
+	bundle exec pod install || bundle exec pod install --repo-update --verbose
+fi
```

Comment:
Code golf. I wonder if something like would work

```
bundle exec pod plugins installed | grep check \
	&& bundle exec pod check
	|| bundle exec pod install \
    || bundle exec  pod install --repo-update --verbose
```

As I said, code golf 🤷‍♂️ It might be DRY but not necessarily cleaner

Reply:
Yeah, I like the DRY of it, but I feel like it'd be a bit harder to maintain?